04c_solara.ipynb

```diff
@@ -6,21 +6,428 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Make the plot\n",
+    "\n",
+    "Either the pyplot or object interface to matplotlib can be used. If pyplot is used, then the plot should be closed after drawing it so that you do not end up with a bunch of open (but inaccessible) plots.\n",
+    "\n",
+    "Since we declared `selected_df` as a reactive variable, the plot is redrawn whenever its value changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| export \n",
+    "\n",
+    "@solara.component\n",
+    "def draw_plot():\n",
+    "    plt.xlabel('Year')\n",
+    "    plt.ylabel('Temperature Anomalies over Land w.r.t. 1951-80 (˚C)')\n",
+    "    plt.title('Global Annual Mean Surface Air Temperature Change')\n",
+    "\n",
+    "    plt.plot(selected_df.value['Year'], selected_df.value['Temperature'], label='Raw Data')\n",
+    "    plt.plot(selected_df.value['Year'], selected_df.value['Smoothed Data'], label='Smoothed Data')\n",
+    "    plt.legend()\n",
+    "    plt.show()\n",
+    "    plt.close()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining the dashboard\n",
+    "\n",
+    "In the cell below we define the dashboard. We could call it anything we want, but when running it as a dashboard using `solara-server` if there is an object called `Page`, then that is what will be rendered in the browser.\n",
+    "\n",
+    "The overall layout will have one row with two columns. The first column has the controls and the second column has the data and graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| export\n",
+    "@solara.component\n",
+    "def Page():\n",
+    "    # These first two components are called here so that solara knows it should call them \n",
+    "    # when changes occur in any of the reactive variables used in those components.\n",
+    "    check_poly_order()\n",
+    "    selected_data()\n",
+    "    \n",
+    "    # We make a row, which will end up with two columns \n",
+    "    with solara.Row():\n",
+    "        # Here we define the left column and restrict its width to 500px.\n",
+    "        with solara.Column(style=dict(width=\"500px\")):\n",
+    "            # Get some extra space at the top...\n",
+    "            solara.Text(\"\\n\\n\")\n",
+    "            # Here we use the controls component we defined above.\n",
+    "            controls()\n",
+    "        # Make column 2 with the data and graph. This column will use whatever space\n",
+    "        # is available that the first column doesn't use.\n",
+    "        with solara.Column():\n",
+    "            # Display the data. The Details component is a collapsible component sort of like\n",
+    "            # an accordian. Its child is an ipydatagrid.DataGrid, like we used previously.\n",
```

Comment:
You misspelt the polkalicious musical instrument:
```suggestion
    "            # an accordion. Its child is an ipydatagrid.DataGrid, like we used previously.\n",
```

Reply:
Fixed